pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -447,19 +457,46 @@ func (m *kubeGenericRuntimeManager) computePodContainerChanges(pod *v1.Pod, podS
 	// check the status of containers.
 	for index, container := range pod.Spec.Containers {
 		containerStatus := podStatus.FindContainerStatusByName(container.Name)
-		if containerStatus == nil || containerStatus.State != kubecontainer.ContainerStateRunning {
+		containerRef, err := kubecontainer.GenerateContainerRef(pod, &container)
```

Comment:
Not sure, maybe we can just discard the error and put the log message in the `GenerateContainerRef` function to begin with?  I'd be good with that.